tests/run-make/asm-incr-cache/rmake.rs

```diff
@@ -0,0 +1,16 @@
+extern crate run_make_support;
+
+use run_make_support::{rustc, tmp_dir};
+
+fn main() {
+    let inc_dir = tmp_dir();
+
+    for _ in 0..=1 {
+        rustc()
+            .input("lib.rs")
+            .arg("--emit=obj,asm,dep-info,link,mir,llvm-ir,llvm-bc")
```

Comment:
Consider introducing some helpers:

```rs
rustc()
    .emit("obj,asm,...")
    .crate_type("lib")
    .incremental(inc_dir) // takes a <P: AsRef<Path>>
```

Reply:
I added helpers here - https://github.com/rust-lang/rust/pull/122597 so it will be fixed after it is merged.

Reply:
Actually, can you move those helper into this PR, as IIUC you would like this PR to be merged before #122597?

Reply:
> Actually, can you move those helper into this PR, as IIUC you would like this PR

Correct, this PR fixes a bug, #122597 introduces new behavior that relies on this bug being fixed. I also asked if new helpers should go somewhere else :) But sure, will move along with all the suggested changes to the API.

Reply:
> I also asked if new helpers should go somewhere else :)

My bad, I thought in the dependent PR by "a separate PR" you meant *yet another PR* and not *this* PR, which is why I said it's fine to keep it in the dependent PR.